coveo-functools/tests_functools/test_flex_deserializer.py

```diff
@@ -313,7 +314,26 @@ def __post_init__(self, change: bool) -> None:
 )
 def test_deserialize_init_var(cls: Any) -> None:
     """Handle a bug with InitVar vs forward references."""
-    assert deserialize({"change": True}, hint=cls).value is True
+    assert deserialize({"change": True}, hint=cls, errors="raise").value is True
+
+
+def test_resolve_hint_init_var() -> None:
+    """
+    The `InitVar` object was not created in a way that plays well with typing's `get_origin` and `get_args`
+    builtins.
+
+    More specifically:
+        - get_origin returns None instead of the InitVar class
+        - get_args returns nothing instead of the type between brackets (str in InitVar[str])
+
+    As such, special handling must be done when we encounter an InitVar instance.
+    """
+    assert resolve_hint(InitVar[int]) == (int, [])
+    assert resolve_hint(InitVar[Optional[int]]) == (Union, [int])

```

Comment:
I don't to much about `InitVar`, but I would have expected

```suggestion
    assert resolve_hint(InitVar[Optional[int]]) == (Optional, [int])

```

Why is my expectation wrong? :thinking:

Reply:
Looking at the implementation, it's still not clear for me how this test passes. My best guess is that `Optional` is implemented as `Union`? :thinking:

Reply:
[Got it on Stack overflow](https://stackoverflow.com/a/51710151/3068259). Still... I'm not sure why `None` is not in the list `[int]`, and I'd prefer to see `Optional` if it is easy enough so that people that read the code don't need to know the implementation details of `Optional`. :sweat_smile:

Reply:
1. Getting rid of `Optional` in favor of `Union[None, ...]` is something the flex machinery has to do at some point. `Optional` is just sugar.
2. Getting rid of `None` is explained [here](https://github.com/coveooss/coveo-python-oss/blob/a2b6f7385b077fa02f92c486e4b4bf43804c06a9/coveo-functools/coveo_functools/flex/helpers.py#L39).

...I feel that the point, is that this should be a private method, but it won't be worth the breaking change.

Reply:
I agree, that's not worth a breaking change. :+1: 

What about adding a comment here so that people not familiar with the details of `coveo-python-oss` that are reading this test are not surprised? :thinking: